handler/schedule.go

```diff
@@ -125,3 +129,43 @@ func (s *ScheduleHandler) mergeResourcesWithDefaults(resources corev1.ResourceRe
 	}
 	return resources
 }
+
+func (s *ScheduleHandler) updateSchedule() error {
+	if err := s.Client.Update(s.CTX, s.schedule); err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error updating resource %s/%s: %w", s.schedule.Namespace, s.schedule.Name, err)
+	}
+	return nil
+}
+
+func (s *ScheduleHandler) updateStatus() error {
+	err := s.Client.Status().Update(s.CTX, s.schedule)
+	if err != nil {
+		s.Log.Error(err, "Could not update SyncConfig.", "name", s.schedule)
+		return err
+	}
+	s.Log.Info("Updated SyncConfig status.")
+	return nil
+}
+
+func (s *ScheduleHandler) getEffectiveSchedule(jobType k8upv1alpha1.JobType, originalSchedule string) string {
+	if s.schedule.Status.EffectiveSchedules == nil {
+		s.schedule.Status.EffectiveSchedules = make(map[k8upv1alpha1.JobType]string)
+	}
+	if existingSchedule, found := s.schedule.Status.EffectiveSchedules[jobType]; found {
+		return existingSchedule
+	}
+	if !strings.HasSuffix(originalSchedule, "-random") {
+		return originalSchedule
+	}
+	schedule := originalSchedule
+	seed := s.schedule.Namespace + "/" + s.schedule.Name + "@" + string(jobType)
```

Comment:
PS: Is it intentional that the seed is deterministic? I.e. for the same namespace, schedule name and jobType it will end up with the same 'random' schedule time over and over again – or did I get this wrong?

Reply:
yes, that's intentional, requested in #116